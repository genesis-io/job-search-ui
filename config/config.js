import development from './development.config';
import production from './production.config';

const environment = 'development';

const config = {
  development,
  production,
};

export default config[environment];
