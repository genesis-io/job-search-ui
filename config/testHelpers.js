import { shallow, mount, render } from 'enzyme';

export const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, { context });
};

export const mountWithStore = (component, store) => {
  const context = {
    store,
  };
  return mount(component, { context });
};

export const renderWithStore = (component, store) => {
  const context = {
    store,
  };
  return render(component, { context });
};

