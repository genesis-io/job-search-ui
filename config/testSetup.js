import './polyFills';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });
// Make Enzyme functions available in all test files without importing
global.shallow = shallow;
global.render = render;
global.mount = mount;

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
};

global.localStorage = new LocalStorageMock;
