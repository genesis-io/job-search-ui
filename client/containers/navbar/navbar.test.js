import React from 'react';
import { createMockStore } from 'redux-test-utils';
import NavBar from './navbar';
import { shallowWithStore } from '../../../config/testHelpers';

let wrapper;
let store;

beforeAll(() => {
  const testState = {
    user: {},
  };
  store = createMockStore(testState);
  wrapper = shallowWithStore(<NavBar />, store);
});

test('render home component', () => {
  expect(wrapper).toMatchSnapshot();
});
