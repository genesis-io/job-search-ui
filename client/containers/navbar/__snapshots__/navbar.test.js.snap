// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home component 1`] = `
<NavBar
  dispatch={[Function]}
/>
`;
